tests/Microsoft.Azure.Batch.SoftwareEntitlement.Tests/FakeCertificateStore.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Batch.SoftwareEntitlement.Common;
+
+namespace Microsoft.Azure.Batch.SoftwareEntitlement.Tests
+{
+    public class FakeCertificateStore : ICertificateStore
+    {
+        private readonly string _notFoundError;
+        private readonly Dictionary<string, X509Certificate2> _certLookup;
```

Comment:
Make this `Dictionary<CertificateThumbprint, X509Certificate2>` - the `CertificateThumbprint` struct implements `.Equals(object)` and `.GetHashCode()` properly, so it should work as a key for the Dictionary.